### PR TITLE
chore(tracing): move tracing out of experimental

### LINKED ISSUE #1325: [traces] move llama_index callback out of experimental
move `llama_index` out of experimental and under `integrations/llama_index` - this callback will be directly used inside of llama_index and be re-exported.

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #1325 

Start moving tracing callbacks and tracers out of experimental

# Steps to move
- [ ] merge to main
- [ ] main to docs
- [ ] publish release

### CODE REVIEW

**Review by axiomofjoy — APPROVED** [review]

**Comment by RogerHYang** [comment]
we can re-export in `__init__` to achieve a shorter import

`from phoenix.trace.llama_index.callback import ...`
vs.
`from phoenix.trace.llama_index import ...`

`from phoenix.trace.langchain.tracer import ...`
vs.
`from phoenix.trace.langchain import ...`

**Comment by mikeldking** [comment]
> `from phoenix.trace.llama_index.callback import ...` vs. `from phoenix.trace.llama_index import ...`
> 
> `from phoenix.trace.langchain.tracer import ...` vs. `from phoenix.trace.langchain import ...`

Yeah good point. I was debating if the longer syntax was better but shorter is probably better.

File: pyproject.toml [hunk 1]
```diff
@@ -56,12 +56,18 @@ dev = [
...
 ]
 experimental = [
-  "langchain>=0.0.257",
-  "llama-index>=0.8.25",
   "openai",
   "tenacity",
 ]
 
+llama_index = [
+  "llama-index>=0.8.25",
```
Comment (RogerHYang) [inline pyproject.toml:64]:
```suggestion
  "openai",
  "llama-index>=0.8.25",
```

Comment (mikeldking) [reply pyproject.toml:64]: @RogerHYang why is OpenAI needed?

Comment (RogerHYang) [reply pyproject.toml:64]: we have a line that says `if isinstance(raw, OpenAIObject):` in there

Comment (RogerHYang) [reply pyproject.toml:64]: maybe we can try to remove that line later

**Review by RogerHYang — APPROVED** [review]

**Comment by RogerHYang** [comment]
just a note: we may have to fail the `import` gracefully, because nothing is preventing the user from calling the tracer without installing the optional dependencies first (the code is all there. it just won't run)

File: pyproject.toml (hunk 1 above)
Comment (mikeldking) [reply pyproject.toml:64]: https://github.com/Arize-ai/phoenix/issues/1388

**Pull request merged**